CMakeLists.txt

```diff
@@ -1055,6 +1055,13 @@ if((MSVC14 AND CMAKE_CL_64) OR ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang"))
 endif()
 
 ################################################################################
+# Some special handling of the compilation is need on build infrastructure for
+# generating packages for target architecture, see issue #3575
+
+hpx_option(HPX_WITH_BUILD_DISTRIBUTION BOOL
+  "Build HPX on the build infrastructure on any LINUX distribution (default: OFF)."
+  OFF ADVANCED)
+
```

Comment:
Wouldn't this be better called `HPX_WITH_BUILD_RPM_DISTRIBUTION`?

Reply:
There could be other non-rpm-based distribution, which want to do the same, e.g. Ubuntu.

Reply:
`HPX_WITH_BUILD_BINARY_PACKAGE`?

Reply:
I like that